src/App.js

```diff
@@ -1,16 +1,64 @@
-import React from 'react';
-import './App.css';
+import React, { useState } from 'react';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+
+import './App.css';
 
 const App = () => {
+  const [messagesData, setMessagesData] = useState(chatMessages);
+
+  const [numLiked, setNumLiked] = useState(0);
+
+  const onHeartClicks = (liked) => {
+    if (liked) {
+      setNumLiked(numLiked - 1);
+    } else {
+      setNumLiked(numLiked + 1);
+    }
+  };
+
+  const onUpdateLikeFx = (id) => {
+    const newMessages = messagesData.map((message) => {
+      if (message.id === id) {
+        return { ...message, liked: !message.liked };
+      } else {
+        return message;
+      }
+    });
+
+    setMessagesData(newMessages);
+  };
+
+  const userOne = messagesData[0]['sender'];
+
+  const findUserTwo = (messagesData) => {
```

Comment:
What would this return if there were no messages yet (or maybe just one sent by the local sender)? Thinking more broadly, what if the conversation originated with the remote sender? (We can't tell from the data we have, but how might we handle this in a more general case?)